src/Registration.vue

```diff
@@ -0,0 +1,162 @@
+ - it under the terms of the GNU Affero General Public License as
+ - published by the Free Software Foundation, either version 3 of the
+ - License, or (at your option) any later version.
+ -
+ - This program is distributed in the hope that it will be useful,
+ - but WITHOUT ANY WARRANTY; without even the implied warranty of
+ - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ - GNU Affero General Public License for more details.
+ -
+ - You should have received a copy of the GNU Affero General Public License
+ - along with this program. If not, see <http://www.gnu.org/licenses/>.
+ -
+ -->
+
+<template>
+	<div id="terms_of_service">
+		<p v-if="terms.length">
+			<input id="terms_of_service_accepted"
+				v-model="hasSigned"
+				:value="termsId"
+				type="checkbox"
+				name="terms_of_service_accepted"
+				class="checkbox">
+			<label
+				for="terms_of_service_accepted"
+				@click.prevent.stop="showTerms">
+				{{ t('terms_of_service', 'I acknowledge that I have read and agree to the above terms of service') }} ↗
+			</label>
+		</p>
+
+		<div id="terms_of_service_confirm">
+			<modal name="confirm-terms"
+				:adaptive="true"
+				@before-close="beforeClose">
+				<div id="tos-overlay">
+					<h3>{{ t('terms_of_service', 'Terms of service') }}</h3>
+					<select v-if="terms.length > 1" v-model="selectedLanguage">
+						<option v-for="(language, index) in languages" :key="index" :value="index">
+							{{ language }}
+						</option>
+					</select>
+					<div class="clear-both" />
+
+					<div class="text-content" v-html="termsBody" />
+
+					<button class="primary"
+						@click.prevent.stop="acceptTerms">
+						{{ t('terms_of_service', 'I acknowledge that I have read and agree to the above terms of service') }}
+					</button>
+				</div>
+			</modal>
+		</div>
+	</div>
+</template>
+
+<script>
+import axios from '@nextcloud/axios'
+import { generateUrl } from '@nextcloud/router'
+import '@nextcloud/dialogs/styles/toast.scss'
+
+export default {
+	name: 'Registration',
+
+	data() {
+		return {
+			hasSigned: false,
+			terms: {},
+			languages: [],
+			selectedLanguage: 0,
+			termsId: 0,
+			termsBody: '',
+			publicContent: null,
+		}
+	},
+
+	watch: {
+		selectedLanguage(newLanguage) {
+			this.selectTerms(newLanguage)
+		},
+	},
+
+	mounted() {
+		this.loadTerms()
+	},
+
+	methods: {
+		async loadTerms() {
+			try {
+				const response = await axios.get(generateUrl('/apps/terms_of_service/terms'))
+
+				// this.hasSigned = response.data.hasSigned
+				this.terms = response.data.terms
+
+				const language = OC.getLanguage().split('-')[0]
+
+				if (!this.terms.length || this.hasSigned) {
+					return
+				}
+
+				// make it Vue
+				this.publicContent = document.getElementById('files-public-content')
+				if (this.publicContent !== null) {
+					this.publicContent.style.visibility = 'hidden'
+				}
+
+				this.selectTerms(0)
+				if (this.terms.length > 1) {
+					Object.keys(this.terms).forEach((index) => {
```

Comment:
strange way of iterating over an array. I believe there are cases where this could cause side effects and return more than just the array keys.

you should be able to use forEach directly on the array https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

Reply:
copied from UserApp.vue but yeah.
Something to clean and deduplicate later